src/main/java/org/jabref/gui/customentrytypes/CustomEntryTypeDialogViewModel.java

```diff
@@ -55,18 +52,14 @@ public Field fromString(String string) {
         }
     };
 
-    private final ListProperty<BibEntryType> entryTypes;
-    private final ListProperty<Field> fields;
-    private final ObjectProperty<BibEntryType> selectedEntryTypes = new SimpleObjectProperty<>();
-    private final ListProperty<FieldViewModel> fieldsForType;
+    private final ObservableList<Field> fieldsForAdding;
+    private final ObjectProperty<CustomEntryTypeViewModel> selectedEntryTypes = new SimpleObjectProperty<>();
     private final ObjectProperty<Field> selectedFieldToAdd = new SimpleObjectProperty<>();
     private final StringProperty entryTypeToAdd = new SimpleStringProperty("");
     private final ObservableList<BibEntryType> allEntryTypes;
-    private final ObservableList<FieldViewModel> allFieldsForType = FXCollections.observableArrayList(extractor -> new Observable[] {extractor.fieldName(), extractor.fieldType()});
     private final ObjectProperty<Field> newFieldToAdd = new SimpleObjectProperty<>();
     private final BibDatabaseMode mode;
-    private final Map<BibEntryType, List<FieldViewModel>> typesWithFields = new HashMap<>();
-    private final List<BibEntryType> typesToRemove = new ArrayList<>();
+    private final ListProperty<CustomEntryTypeViewModel> entryTypesWithFields = new SimpleListProperty<>(FXCollections.observableArrayList(extractor -> new Observable[] {extractor.entryType(), extractor.fields()}));
```

Comment:
observable list should be sufficient here.

Reply:
Nope, it's used in a binding and bind() only accepts a ListProperty

Reply:
`entryTypes.setItems(viewModel.entryTypes());` should work if `entryTypes` is an observable list. No need for the bind + property.